### ISSUE TITLE
Error running miktex on Win 10 ver. 1709

### ISSUE THREAD

**Issue by cwtex4110** [MiKTeX/miktex#41]
# Bug report
I apology that this is really not a bug report of MiKTeX, 
but a report on how MiKTeX is affected by Windows 10 ver. 1709 (OS build 16299.98).
I am asking for help.

I have developed a Chinese TeX system called cwTeX:

http://homepage.ntu.edu.tw/~ntut019/cwtex/cwtex.html

The system is a preprocessor, built on MiKTeX, plus some Chinese fonts.
It has been used to typeset many textbooks and papers in Taiwan.

A mwe is provided in the following.
The line before the last one contains 7 Chinese characters,
coming from 5 Chinese fonts (each contains 256 characters).
To typeset this file, one has to run the cwtex preprocessor first, 
and then run pdflatex (or latex).

Before Windows 10 ver. 1709 (Oct. 2017), there is no problem 
running this file.  Win 7 and 8 are fine. So are old versions of MiKTeX.
However, the most recent update of Windows 10 (ver. 1709) created an error,
with the following message:

2017-12-09 21:58:48,734+0800 FATAL pdflatex - Line: 477
2017-12-09 22:03:01,455+0800 FATAL pdflatex - No space left on device
2017-12-09 22:03:01,455+0800 FATAL pdflatex - Info:
2017-12-09 22:03:01,455+0800 FATAL pdflatex - Source: libraries\miktex\texandfriends\include\miktex\c4p\C4P.h

Now if I deleted one or two Chinese characters in the mwe, it will run fine.
Or if I change  \fontsize{120}{40pt}  to  \fontsize{100}{40pt},
it will also run fine.

I have tried this mwe on several computers, with different hardware,
and only one computer (8gb RAM) can avoid this problem.
It does not seem to be related to the amount of RAM,
because one computer with 16gb RAM has problem.
All the computers are installed with the most recent version of MiKTeX.

Before Microsoft fixes the issue in the future, I wonder if anything can be
done or tried, say adding options when running pdflatex, to fix the problem (maybe temporarily)?

If needed, I can send the fonts needed, or the whole system can be downloaded
form the cwTeX webpage above.

Thanks!

Tsong-Min Wu
Department of Economics

********** mwe ************
\documentclass[12pt]{book}
\usepackage{charter}
\usepackage{graphicx,xcolor}
\usepackage{titlesec}

\newcommand{\numtitle}{
 \titleformat{\chapter}[hang]{\sf\it}%
  {\fontsize{120}{40pt}\selectfont\textcolor{blue}\thechapter}%
  {1pc}{\fontsize{25}{20pt}\selectfont}
  [{\vspace{1.5pc}\titlerule[1.5pt]\vspace{-1.5pc}}]}

\begin{document}
\fontsize{12}{21pt}\selectfont
\setcounter{chapter}{13}

\numtitle
\chapter{\textbf{Ramsey} 模型的均衡分析}
\end{document}
***************************

**Comment by edocevoli** [comment]
The error occurs while writing a character into a text output file.
(Line 477 in `C4P.h` refers to the `putc` function.)
There are only two text output files: `mwe.aux` and `mwe.log`.

**Comment by cwtex4110** [comment]
I have tried using cwTeX + texlive2017 to run the mwe, 
and there is no error.

Tsong-Min Wu

**Comment by edocevoli** [comment]
I cannot reproduce this issue and I do not have enough information to locate/fix the bug.

Running `pdflatex mwe.tex` on a Windows 10 1709 creates no error.

**Comment by edocevoli** [comment]
See also: https://tex.stackexchange.com/questions/408162/fatal-pdflatex-no-error

**Comment by cwtex4110** [comment]
Instead of pdflatex and latex, I have tried to run the mwe.tex above,
and also other documents, and it runs successfully. 

Hope this helps.

Tsong-Min Wu

**Comment by u-fischer** [comment]
Can you show the output of the preprocessing?

**Comment by u-fischer** [comment]
I wonder if this problems (see also the new report about xelatex crashing in the mailing list) has to do with broken utf8 in some output files. ShreevatsaR showed here https://tex.stackexchange.com/a/407793/2388 that xelatex can write invalid utf8, pdflatex does it anyway but lualatex don't.

**Comment by FYPetro** [comment]
I am writing a document in Chinese and I have got the same error. Here is the log.
```
2018-06-20 11:34:51,642+0800 INFO  pdflatex - starting with command line: pdflatex scila.tex -synctex=-1 -interaction=nonstopmode -shell-escape
2018-06-20 11:34:51,650+0800 INFO  pdflatex - allowing known shell commands
2018-06-20 11:34:51,650+0800 INFO  pdflatex - enabling input (output) from (to) processes
2018-06-20 11:34:51,651+0800 INFO  pdflatex - allowing all shell commands
2018-06-20 11:34:51,682+0800 INFO  pdflatex - UTF8 BOM detected: scila.tex
2018-06-20 11:34:51,684+0800 INFO  pdflatex - UTF8 BOM detected: scila.cls
2018-06-20 11:34:51,684+0800 INFO  pdflatex - UTF8 BOM detected: scila.cls
2018-06-20 11:34:53,029+0800 INFO  pdflatex - UTF8 BOM detected: scilamath.tex
2018-06-20 11:34:53,029+0800 INFO  pdflatex - UTF8 BOM detected: scilamath.tex
2018-06-20 11:34:53,370+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear.tex
2018-06-20 11:34:53,370+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear.tex
2018-06-20 11:34:53,391+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/vector-spaces.tex
2018-06-20 11:34:53,391+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/vector-spaces.tex
2018-06-20 11:34:53,395+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/bases-and-dimension.tex
2018-06-20 11:34:53,396+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/bases-and-dimension.tex
2018-06-20 11:34:53,399+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/linear-transformations.tex
2018-06-20 11:34:53,399+0800 INFO  pdflatex - UTF8 BOM detected: tex/linear/linear-transformations.tex
2018-06-20 11:34:53,421+0800 FATAL pdflatex - No space left on device
2018-06-20 11:34:53,421+0800 FATAL pdflatex - Info: 
2018-06-20 11:34:53,421+0800 FATAL pdflatex - Source: libraries\miktex\texandfriends\include\miktex\c4p\C4P.h
2018-06-20 11:34:53,421+0800 FATAL pdflatex - Line: 477
2018-06-20 11:34:53,426+0800 INFO  pdflatex - finishing with exit code 1
```

**Comment by jcchu** [comment]
I was able to get it working by first changing the console’s code page to UTF-8.  Try executing `chcp 65001` before `xelatex` and see if it solves your problem.  @tigerhawkvok @magnus-madsen @cwtex4110

**Comment by garylavayou** [comment]
I tested the MWE with MikTeX and found that it cannot be successfully compiled unless I add the CJK package.
```latex
\usepackage{cjkutf8}
\begin{CJK}{UTF8}{gbsn}
  \numtitle
  \chapter{\textbf{Ramsey} 模型的均衡分析}
\end{CJK}
```
---
Besides, I also confronted a problem with the same log information, when using MikTeX to compile a Chinese thesis.

**Comment by olliechick** [comment]
I fixed this by closing the PDF that I was trying to write to.

**Comment by edocevoli** [comment]
This has been fixed in MiKTeX 20.6: programs will switch console output code page to 65001 (UTF-8) in order to display  Unicode characters.
